scrapy/dispatch/weakref_backports.py

```diff
@@ -0,0 +1,67 @@
+"""
+weakref_backports is a partial backport of the weakref module for python
+versions below 3.4.
```

Comment:
I'm sorry, as usual I spoke too soon. That implementation backports `weakref.finalize`. We ported `weakref.WeakMethod`. In the current state of that package we would require both this and that package.